### ISSUE TITLE
collinearity_detection has a error when training if using dask distrubuted

### ISSUE THREAD

**Issue by SonixLegend** [DataCanvasIO/HyperGBM#94]
<em>Please make sure that this is a bug. </em>

**System information**
- OS Platform and Distribution (e.g., CentOS 7.6): CentOS 7.9 Docker 20.10.14
- Python version: 3.9
- HyperGBM version: 0.2.5.4
- Other Python packages(run `pip list`): 
        anyio==3.6.1
        argon2-cffi==21.3.0
        argon2-cffi-bindings==21.2.0
        asttokens==2.0.5
        attrs==22.1.0
        Babel==2.10.3
        backcall==0.2.0
        bcrypt==3.2.2
        beautifulsoup4==4.11.1
        bleach==5.0.1
        bokeh==2.4.3
        catboost==1.0.6
        certifi==2022.6.15
        cffi==1.15.1
        charset-normalizer==2.1.0
        click==8.1.3
        cloudpickle==2.1.0
        convertdate==2.4.0
        cryptography==37.0.4
        cycler==0.11.0
        dask==2022.7.1
        dask-glm==0.2.0
        dask-ml==2022.5.27
        debugpy==1.6.2
        decorator==5.1.1
        defusedxml==0.7.1
        distributed==2022.7.1
        entrypoints==0.4
        executing==0.9.1
        fastjsonschema==2.16.1
        featuretools==1.12.0
        fonttools==4.34.4
        fsspec==2022.7.1
        graphviz==0.20.1
        hboard==0.1.0
        hboard-widget==0.1.0
        HeapDict==1.0.1
        hijri-converter==2.2.4
        holidays==0.14.2
        hypergbm==0.2.5.4
        hypernets==0.2.5.4
        idna==3.3
        imbalanced-learn==0.9.1
        importlib-metadata==4.12.0
        iniconfig==1.1.1
        ipykernel==6.15.1
        ipython==8.4.0
        ipython-genutils==0.2.0
        ipywidgets==7.7.1
        jedi==0.18.1
        jieba==0.42.1
        Jinja2==3.1.2
        joblib==1.1.0
        json5==0.9.9
        jsonschema==4.9.0
        jupyter-client==7.3.4
        jupyter-core==4.11.1
        jupyter-server==1.18.1
        jupyterlab==3.4.4
        jupyterlab-pygments==0.2.2
        jupyterlab-server==2.15.0
        jupyterlab-widgets==1.1.1
        kiwisolver==1.4.4
        korean-lunar-calendar==0.2.1
        lightgbm==3.3.2
        llvmlite==0.39.0
        locket==1.0.0
        MarkupSafe==2.1.1
        matplotlib==3.5.2
        matplotlib-inline==0.1.3
        mistune==0.8.4
        msgpack==1.0.4
        multipledispatch==0.6.0
        nbclassic==0.4.3
        nbclient==0.6.6
        nbconvert==6.5.0
        nbformat==5.4.0
        nest-asyncio==1.5.5
        notebook==6.4.12
        notebook-shim==0.1.0
        numba==0.56.0
        numpy==1.22.4
        packaging==21.3
        pandas==1.4.3
        pandocfilters==1.5.0
        paramiko==2.11.0
        parso==0.8.3
        partd==1.2.0
        pexpect==4.8.0
        pickleshare==0.7.5
        Pillow==9.2.0
        plotly==5.9.0
        pluggy==1.0.0
        prettytable==3.3.0
        prometheus-client==0.14.1
        prompt-toolkit==3.0.30
        psutil==5.9.1
        ptyprocess==0.7.0
        pure-eval==0.2.2
        py==1.11.0
        pyarrow==8.0.0
        pycparser==2.21
        Pygments==2.12.0
        PyMeeus==0.5.11
        PyNaCl==1.5.0
        pyparsing==3.0.9
        pyrsistent==0.18.1
        pytest==7.1.2
        python-dateutil==2.8.2
        python-geohash==0.8.5
        pytz==2022.1
        PyYAML==6.0
        pyzmq==23.2.0
        requests==2.28.1
        scikit-learn==1.1.1
        scipy==1.9.0
        seaborn==0.11.2
        Send2Trash==1.8.0
        shap==0.41.0
        six==1.16.0
        slicer==0.0.7
        sniffio==1.2.0
        sortedcontainers==2.4.0
        soupsieve==2.3.2.post1
        stack-data==0.3.0
        tblib==1.7.0
        tenacity==8.0.1
        terminado==0.15.0
        threadpoolctl==3.1.0
        tinycss2==1.1.1
        tomli==2.0.1
        toolz==0.12.0
        tornado==6.1
        tqdm==4.64.0
        traitlets==5.3.0
        typing_extensions==4.3.0
        urllib3==1.26.11
        wcwidth==0.2.5
        webencodings==0.5.1
        websocket-client==1.3.3
        widgetsnbextension==3.6.1
        woodwork==0.17.1
        xgboost==1.6.1
        XlsxWriter==3.0.3
        zict==2.2.0
        zipp==3.8.1

**Describe the current behavior**
We wanna train a dataset via dask distrubuted with 3 nodes.

**Describe the expected behavior**
Training finished successfully

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Jupyter notebook.
[dask.zip](https://github.com/DataCanvasIO/HyperGBM/files/9266431/dask.zip)

**Are you willing to submit PR?(Yes/No)**
Yes

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. 
If including tracebacks, please include the full traceback. Large logs and files 
should be attached.
![企业微信截图_16595067253520](https://user-images.githubusercontent.com/26000465/183026601-27068b37-8ddc-4303-a6a3-f4da3f35b409.png)
![企业微信截图_16595067711352](https://user-images.githubusercontent.com/26000465/183026648-6f575e18-7825-4d51-9e4b-13cf1a185d02.png)
![企业微信截图_16595068094460](https://user-images.githubusercontent.com/26000465/183026677-6143418a-bdac-4edd-ab81-e171026c4466.png)

**Comment by oaksharks** [comment]
Hi SonixLegend, thanks for your report.
The issue has fixed in this commit: https://github.com/DataCanvasIO/Hypernets/commit/f11327e1d1acdf6faead5a5d3845384326bb26d3 , you can update Hypernets and and try again.

**Comment by SonixLegend** [comment]
Thanks, We can run it.
